### PR TITLE
hashtable

### PR DESCRIPTION
**Pull request by sameeksharl**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes

### Description
 hash table implemented in c which includes all CRUD operations (insert, display, update , delete) 
 and also included to return the size of table instantly.
(here updating happens if  the key already present in table )

### Checklist

- [ yes] I've read the contribution guidelines.
- [yes ] I've checked the issue list before deciding what to submit.
- [ yes] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)

### CODE REVIEW

File: C/ds/hashtable.c [hunk 1]
```diff
@@ -0,0 +1,222 @@
+// Implementing hash table in C
```
Comment (bhav09) [inline C/ds/hashtable.c:1]: Description at the start of the code file.

File: C/ds/hashtable.c [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+{
+  int key;
+  int data;
+  int collision;
+};
+struct set *array;
+int capacity = 10;
+int size = 0;
+
+int hashFunction(int key)
```
Comment (bhav09) [inline C/ds/hashtable.c:16]: Function descriptions and thus comments are lacking.

File: C/ds/hashtable.c [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+  array = (struct set *)malloc(capacity * sizeof(struct set));
+  for (int i = 0; i < capacity; i++)
+  {
+  array[i].key = 0;
+  array[i].data = 0;
+  array[i].collision = 0;
+  }
+}
+
+void insert(int key, int data)
```
Comment (bhav09) [inline C/ds/hashtable.c:61]: Make sure the comment is above the line of code and not alongside.

File: C/ds/hashtable.c [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+  array[index].data = data;
+  size++;
+   printf("\n Collision occured  \n\n");
+   printf("\n Key (%d) has been inserted at %d using linear probing \n\n", key,index);
+  }
+  else if (array[index].key == key)
+  {
+  array[index].data = data;
+  }
+
```
Comment (bhav09) [inline C/ds/hashtable.c:98]: Remove redundant spaces. that are in the form of new lines.

File: C/ds/hashtable.c [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+  default:
+
+    exit(0);
+  }
+
+  
+
+  } while (1);
+}
```
Comment (bhav09) [inline C/ds/hashtable.c:222]: At the end of the code file, write the complexity and sample I/O.

**Review by bhav09 — CHANGES_REQUESTED** [review]
Please work on the following areas mentioned in the review.

**Review by ankitaggarwal23 — COMMENTED** [review]
Code already present

**Pull request closed without merging**